src/ascii_fold.rs

```diff
@@ -0,0 +1,1534 @@
+/*
+* Much of this file is either directly copied or inspired by the ascii_folding_filter
+* code in tantivy:
+* https://github.com/tantivy-search/tantivy/blob/main/src/tokenizer/ascii_folding_filter.rs
+* That code in turn is adapted from the similar code in Lucene:
+* // https://github.com/apache/lucene-solr/blob/master/lucene/analysis/common/src/java/org/apache/lucene/analysis/miscellaneous/ASCIIFoldingFilter.java
+* The Tantivy code is licensed under the MIT License and it is copyright by
+* the AUTHORS of tantivy.
+* Code copied on 2021-09-10. File last edited on 2021-07-01.
+* Each function is individually labeled as well.
+*/
+
+// This function is inspired by the to_ascii function in
+// tantivy-search's ascii_folding_filter
+pub fn latinize(text: &str) -> String {
+    let mut output = String::with_capacity(text.len());
+    for c in text.chars() {
+        if let Some(folded) = fold_non_ascii_char(c) {
+            output.push_str(folded);
+        } else {
+            output.push(c);
+        }
+    }
+    output
+}
```

Comment:
I feel like an `if let` that also has an `else` makes more sense just as a match. I checked out what this was inspired by so I understand if you'd prefer to leave it closer to that, but to me it would make more sense as
```
    for c in text.chars() {
        match fold_non_ascii_char(c) {
            Some(modified) => output.push_str(modified),
            None => output.push(c),
        }
    }
```

Reply:
Totally agree. Fixed.